### PR TITLE
add(test): test case to fetch specific snapshot information using target

### LINKED ISSUE #2936: rebuild estimate for cstor replica
## Description
As of now CVR has a status field which describes replica health status and rebuilding information, i.e if it is rebuilding or not. But it doesn't say much about how much data it is missing from healthy replica or how much time it will take to become healthy.
If such information is added to CVR then it will help the users to get more idea about what is happening in rebuild process and the amount of time it will take to complete rebuilding.

## Possible Solution
CVR with existing/missing snapshot  details, plus rebuilding progress

### PR DESCRIPTION
**Pull request by mynktl**
fixes: openebs/openebs#2936
Signed-off-by: mayank <mayank.patel@mayadata.io>

### CODE REVIEW

**Comment by mynktl** [comment]
depends on https://github.com/openebs/libcstor/pull/46

File: tests/cstor/gtest/test_zrepl_prot.cc [hunk 1]
```diff
@@ -2020,7 +2045,31 @@ TEST(Snapshot, CreateAndDestroy) {
...
+	buf = (char *)malloc(hdr_in.len);
+	rc = read(control_fd, buf, hdr_in.len);
+	ASSERT_EQ(rc, hdr_in.len);
+	snaplist = (struct zvol_snapshot_list *) buf;
+	output = execCmd("zfs", std::string("get guid -Hpo value ") +
+	    vol_name);
+	EXPECT_EQ(snaplist->zvol_guid, std::stoul(output));
+	verify_snapshot_details(vol_name, snaplist->data, "");
+
+	// Try to fetch snapshot list using snapshot name
```
Comment (pawanpraka1) [inline tests/cstor/gtest/test_zrepl_prot.cc:2059]: add one more test to send wrong name in the snaplist command.

Comment (mynktl) [reply tests/cstor/gtest/test_zrepl_prot.cc:2059]: done

**Review by vishnuitta — APPROVED** [review]
changes are good

**Review by pawanpraka1 — APPROVED** [review]
looks good. Have megred the libcstor PR and started the travis build for this. Will merge it once travis passes.

**Pull request merged**